generators/client/templates/src/main/webapp/app/blocks/config/_compile.config.js

```diff
@@ -5,13 +5,15 @@
         .module('<%=angularAppName%>')
         .config(compileServiceConfig);
 
-    compileServiceConfig.$inject = ['$compileProvider', 'ENV'];
+    compileServiceConfig.$inject = ['$compileProvider'];
 
-    function compileServiceConfig($compileProvider, ENV) {
+    function compileServiceConfig($compileProvider) {
         // disable debug data on prod profile to improve performance
-        if(ENV === 'prod'){
-            $compileProvider.debugInfoEnabled(false);
-        }
+        
+        // TODO: Howto fetch the dynamic profile in this config.... can this be done elsewhere ?
```

Comment:
@deepu105  : The compileProvider can only be accessed in config block.

You can pull the compileProvider out of the config block and use it in the run method.

```
(function() {
    'use strict';

    var provider = {};

    angular
        .module('ribbon8App', [
            'ngStorage', 
            'ngResource',
            'ngCookies',
            'ngAria',
            'ngCacheBuster',
            'ngFileUpload',
            'ui.bootstrap',
            'ui.bootstrap.datetimepicker',
            'ui.router',
            'infinite-scroll',
            // jhipster-needle-angularjs-add-module JHipster will add new module here
            'angular-loading-bar'
        ])
        .config(config)
        .run(run);

    config.$inject = ['$compileProvider'];

    function config($compileProvider) {
        angular.copy($compileProvider, provider);
    }

    run.$inject = ['stateHandler','ActiveProfiles'];

    function run(stateHandler,ActiveProfiles) {
        stateHandler.initialize();
        
        ActiveProfiles.activeProfiles().then(function(response) {
            provider.debugInfoEnabled(!response.inProduction);
        });        
    }
})();
```